### PR TITLE
SECURITY-3153: Use the Item as a Scope instead of System Scope.

### PR DESCRIPTION
**Pull request by phreakadelle**
### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java [hunk 1]
```diff
@@ -67,22 +80,50 @@ public AbstractMavenArtifactChoiceListProvider(final String artifactId) {
 
     @Override
     public List<String> getChoiceList() {
-
+    	StaplerRequest req = Stapler.getCurrentRequest();
+    	Item item = req.findAncestorObject(Item.class);
```
Comment (yaroslavafenkin) [inline src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:84]: See https://github.com/jenkinsci/stapler/blob/master/core/src/main/java/org/kohsuke/stapler/AncestorInPath.java

File: src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java [hunk 2]
```diff
@@ -67,22 +80,50 @@ public AbstractMavenArtifactChoiceListProvider(final String artifactId) {
...
         
         // FIXME: CHANGE-1: Return only the keys, that are shorter then the values
         // return new ArrayList<String>(mChoices.keySet());
         return new LinkedList<String>(mChoices.values());
     }
 
+    @Extension
+    public static class DescriptorImpl extends Descriptor<ChoiceListProvider> {
+
+		public ListBoxModel doFillCredentialsIdItems(Item item, String credentialsId) {
```
Comment (yaroslavafenkin) [inline src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:100]: Does this work without annotations?
```suggestion
		public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
```

File: src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java [hunk 3]
```diff
@@ -91,9 +132,10 @@ public List<String> getChoiceList() {
      *            the internal jenkins id for the credentials
      * @return the credentials for the ID or NULL
      */
-    public static UsernamePasswordCredentialsImpl getCredentials(final String pCredentialId) {
-        return CredentialsMatchers.firstOrNull(
-                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement> emptyList()),
+    public static UsernamePasswordCredentialsImpl getCredentials(@Nonnull String pCredentialId, @Nonnull Item pItem) {
+		final Authentication acl = pItem instanceof Queue.Task ? Tasks.getAuthenticationOf((Queue.Task) pItem) : ACL.SYSTEM;
+		return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, Jenkins.get(), acl, Collections.<DomainRequirement> emptyList()),
```
Comment (yaroslavafenkin) [inline src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:138]: It's not the Authentication that's passed that is a problem, but the context.
```suggestion
                CredentialsProvider.lookupCredentials(UsernamePasswordCredentialsImpl.class, pItem, ACL.SYSTEM, Collections.<DomainRequirement> emptyList()),
```

File: src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java (hunk 1 above)
Comment (phreakadelle) [reply src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:84]: Hey @yaroslavafenkin,

i think its not possible to use the @AnchestorInPath annotation here, as i have to override/implement get getChoiceList() method from the superclass.

```
@Override
public List<String> getChoiceList() {
    StaplerRequest req = Stapler.getCurrentRequest();
    Item item = req.findAncestorObject(Item.class);
    ....
```

Comment (yaroslavafenkin) [reply src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:86]: Ah, you're right, sorry. Then your current approach works

Comment (phreakadelle) [reply src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:86]: Are we good to build a new version?

Comment (yaroslavafenkin) [reply src/main/java/org/jenkinsci/plugins/maven_artifact_choicelistprovider/AbstractMavenArtifactChoiceListProvider.java:86]: Let me re-test the fix after the recent changes

**Review by yaroslavafenkin — APPROVED** [review]
Re-tested, seems fine now.

**Pull request merged**